etc/kayobe/ansible/growroot.yml

```diff
@@ -27,57 +27,60 @@
-        update_cache: "{{ True if os_family == 'Debian' else omit }}"
-      become: True
-      when: growpart_check.rc != 0
+    - block:
+        - name: Ensure growpart is installed
+          package:
+            name: "{% if os_family == 'RedHat' %}cloud-utils-growpart{% else %}cloud-guest-utils{% endif %}"
+            state: present
+            cache_valid_time: "{{ apt_cache_valid_time if os_family == 'Debian' else omit }}"
+            update_cache: "{{ true if os_family == 'Debian' else omit }}"
+          become: true
 
-    - name: Get root PV device
-      command: "pvs --select vg_name={{ growroot_vg }} --reportformat json"
-      register: pvs
-      become: True
-      changed_when: False
-      check_mode: false
+        - name: Get root PV device
+          command: "pvs --select vg_name={{ growroot_vg }} --reportformat json"
+          register: pvs
+          become: true
+          changed_when: false
+          check_mode: false
 
-    - name: Fail if root PV device not found
-      fail:
-        msg: >
-          Expected LVM physical volume devices not found in volume group {{ growroot_vg }}
-      when: (pvs.stdout | from_json).report[0].pv | length == 0
+        - name: Fail if root PV device not found
+          fail:
+            msg: >
+              Expected LVM physical volume devices not found in volume group {{ growroot_vg }}
+          when: (pvs.stdout | from_json).report[0].pv | length == 0
 
-    - name: Grow partition
-      command: "growpart {{ disk }} {{ part_num }}"
-      vars:
-        pv: "{{ pvs.stdout | from_json }}"
-        disk_tmp: "{{ pv.report[0].pv[0].pv_name[:-1] }}"
-        disk: "{{ disk_tmp[:-1] if disk_tmp[-1] == 'p' else disk_tmp }}"
-        part_num: "{{ pv.report[0].pv[0].pv_name[-1] }}"
-      become: True
-      failed_when: "growpart.rc != 0 and 'NOCHANGE' not in growpart.stdout"
-      changed_when: "'NOCHANGE' not in growpart.stdout"
-      register: growpart
+        - name: Grow partition
+          command: "growpart {{ disk }} {{ part_num }}"
+          vars:
+            pv: "{{ pvs.stdout | from_json }}"
+            disk_tmp: "{{ pv.report[0].pv[0].pv_name[:-1] }}"
+            disk: "{{ disk_tmp[:-1] if disk_tmp[-1] == 'p' else disk_tmp }}"
+            part_num: "{{ pv.report[0].pv[0].pv_name[-1] }}"
+          become: true
+          failed_when: "growpart.rc != 0 and 'NOCHANGE' not in growpart.stdout"
+          changed_when: "'NOCHANGE' not in growpart.stdout"
+          register: growpart
 
-    - name: Grow LVM PV
-      command: "pvresize {{ disk }}"
-      vars:
-        pv: "{{ pvs.stdout | from_json }}"
-        disk: "{{ pv.report[0].pv[0].pv_name }}"
-      become: True
+        - name: Grow LVM PV
+          command: "pvresize {{ disk }}"
+          vars:
+            pv: "{{ pvs.stdout | from_json }}"
+            disk: "{{ pv.report[0].pv[0].pv_name }}"
+          become: true
+      when: lvm_check.rc != 1 or growroot_skip_lvm_check
```

Comment:
Why 1? Would it be safer to check `lvm_check.rc == 0`?

Also, should it be `and not growroot_skip_lvm_check | bool`?

Reply:
Right, but based on the name we want to skip when it's true, or not skip when it's false

Reply:
Yep, you're right. I should really stop contradicting you at 5PM